index.php

```diff
@@ -20,7 +20,7 @@
 
 	if(array_key_exists('action', $params)) {
 		$action = $actionFac->create($params['action']);
-		$action->setUserData($params);
+		$action->userData = $params;
```

Comment:
This will be an issue with [newer PHP versions](https://wiki.php.net/rfc/deprecate_dynamic_properties) for actions that do not have the property.

Reply:
Will keep that mind mind. Will probably refactor this so that input is passed to `execute()` method of actions.